### PR TITLE
Change displayed_on_eshop to function field

### PR DESCRIPTION
**Pull request by priyankarani**

### CODE REVIEW

File: product.py [hunk 1]
```diff
@@ -72,6 +76,37 @@ def __setup__(cls):
...
         })
 
+        # Change displayed_on_eshop to function field
+        cls.displayed_on_eshop = fields.Function(
+            fields.Boolean('Displayed on E-Shop?'),
+            'get_displayed_on_eshop', searcher="search_displayed_on_eshop"
+        )
+
+    @classmethod
+    def search_displayed_on_eshop(cls, name, domain):
```
Comment (tarunbhardwaj) [inline product.py]: This will slow down everything.

File: product.py [hunk 2]
```diff
@@ -72,6 +76,37 @@ def __setup__(cls):
...
+
+    def _get_webshop_listing(self):
+        """
+        Returns webshop listing for current product
+        """
+        ProductListing = Pool().get('product.product.channel_listing')
+
+        return ProductListing.search([
+            ('product', '=', self.id),
+            ('channel.source', '=', 'webshop')
```
Comment (tarunbhardwaj) [inline product.py]: we expect here either one webshop or all webshop will have same product.

What is we use channel from context(current_channel) or `nereid_website.channel`

File: product.py [hunk 3]
```diff
@@ -283,3 +318,46 @@ def availability(cls, uri):
...
+            'required': Eval('channel_source') == 'webshop',
+            'invisible': Eval('channel_source') != 'webshop',
+        }, depends=['channel_source']
+    )
+
+
+class AddProductListingStart:
+    __name__ = 'product.listing.add.start'
+
+    uri = fields.Char(
```
Comment (tarunbhardwaj) [inline product.py]: defaults to product.code

File: tests/test_product.py [hunk 4]
```diff
@@ -86,7 +86,11 @@ def _create_product_template(self, name, vlist, uri, uom=u'Unit'):
             values['products'] = [
                 ('create', [{
                     'uri': uri,
-                    'displayed_on_eshop': True
+                    'channel_listings': [
+                        ('create', [{
+                            'channel': self.webshop_channel,
```
Comment (tarunbhardwaj) [inline tests/test_product.py]: uri on listing ?

File: tests/test_product.py [hunk 5]
```diff
@@ -609,12 +627,19 @@ def test_0050_product_serialize(self):
                 'media': [('create', [{
                     'static_file': file1.id,
                 }])],
+                'products': [
+                    ('create', [{
+                        'uri': 'uri',
+                        'channel_listings': [
+                            ('create', [{'channel': self.webshop_channel}])
```
Comment (tarunbhardwaj) [inline tests/test_product.py]: uri on listing

File: view/product_form_nereid.xml [hunk 6]
```diff
@@ -13,4 +13,12 @@
             <field name="start_displaying_available_quantity"/>
         </page>
     </xpath>
+
+    <xpath expr="/form/label[@name=&quot;active&quot;]"
+            position="before">
```
Comment (tarunbhardwaj) [inline view/product_form_nereid.xml]: fix indentation

**Comment by tarunbhardwaj** [comment]
Do not forget to make changes in nereid-catalog here
https://github.com/fulfilio/nereid-catalog/blob/develop/product.py

File: product.py (hunk 1 above)
Comment (priyankarani) [inline product.py]: fixed

File: product.py (hunk 2 above)
Comment (prakashpp) [inline product.py]: This should respect current channel, if not there look for website channel.

File: product.py [hunk 7]
```diff
@@ -72,6 +77,35 @@ def __setup__(cls):
...
 
+        # Change displayed_on_eshop to function field
+        cls.displayed_on_eshop = fields.Function(
+            fields.Boolean('Displayed on E-Shop?'),
+            'get_displayed_on_eshop', searcher="search_displayed_on_eshop"
+        )
+
+    @classmethod
+    def search_displayed_on_eshop(cls, name, domain):
+        products = cls.search([
```
Comment (prakashpp) [inline product.py]: this should also look for current channel of type webshop

File: product.py [hunk 8]
```diff
@@ -72,6 +77,35 @@ def __setup__(cls):
             ),
         })
 
+        # Change displayed_on_eshop to function field
+        cls.displayed_on_eshop = fields.Function(
+            fields.Boolean('Displayed on E-Shop?'),
+            'get_displayed_on_eshop', searcher="search_displayed_on_eshop"
+        )
+
```
Comment (prakashpp) [inline product.py]: Also make uri a function field

File: product.py [hunk 9]
```diff
@@ -72,6 +77,35 @@ def __setup__(cls):
...
+    @classmethod
+    def search_displayed_on_eshop(cls, name, domain):
+        products = cls.search([
+            ('channel_listings.channel.source', '=', 'webshop')
```
Comment (tarunbhardwaj) [inline product.py]: add current_channel in domain

**Pull request closed without merging**